### PR TITLE
[dv/alert_handler] update the ping tests to hit more corner cases

### PR DESCRIPTION
**Pull request by cindychip**
The original `aler_handler_ping_rsp_vseq` will enable the ping timeout
failure and let stress_all and reset tests try to randomly hit all the
corner cases. But some issues for this tests:
1). Long run time. Usually around 2 hours.
2). Could hardly hit any corner cases:
    a). ping interrupted by a reset
    b). ping interrupted by real escalation signal

To try to cover all the corner cases with a reasonable amount of time,
this PR:
1). Modify the ping_rsp_fail vseq to intentionally hit the corner cases,
and to avoid conditions where ping_failure and check_interrupts happened
in the same cycle (this makes scb very hard to predict)
2). Modify LFSR input `wait_cyc_mask_i` to force design to send out ping
request more frequently

The following PR will add a functional coverage for all alert and
escalator to ensure they receiver a ping request. And this coverage will
be disabled when running the `alert_handler_ping_corner_cases` test.

Signed-off-by: Cindy Chen <chencindy@google.com>

### CODE REVIEW

**Comment by cindychip** [comment]
This PR carries on the discussion on PR: #3115. I was not able to reopen the PR after a wrong force push.. Sorry for the inconvenience.

**Comment by cindychip** [comment]
The latest push solved the merging conflict.

File: hw/ip/alert_handler/dv/tb/tb.sv [hunk 1]
```diff
@@ -82,6 +82,9 @@ module tb;
   );
 
   initial begin
+    static bit reduce_ping_timer_wait_cycles = 0;
```
Comment (weicaiyang) [inline hw/ip/alert_handler/dv/tb/tb.sv:85]: is `static` needed?

File: hw/ip/alert_handler/dv/env/seq_lib/alert_handler_sanity_vseq.sv [hunk 2]
```diff
@@ -81,88 +81,98 @@ class alert_handler_sanity_vseq extends alert_handler_base_vseq;
...
+      if (do_esc_intr_timeout) begin
+        cfg.clk_rst_vif.wait_clks(max_intr_timeout_cyc);
+        // this task checks three sets of registers related to alert/esc status:
+        // alert_accum_cnt, esc_cnt, class_state
+        read_esc_status();
+        // only check interrupt when no esc_int_err, otherwise clear interrupt might happen the
+        // same cycle as interrupt triggered by esc_int_err
+      end
+      // only check interrupt when no esc_int_err, otherwise clear interrupt might happen the
+      // same cycle as interrupt triggered by esc_int_err
```
Comment (weicaiyang) [inline hw/ip/alert_handler/dv/env/seq_lib/alert_handler_sanity_vseq.sv:157]: nit: repetitive comments

File: hw/ip/alert_handler/dv/env/seq_lib/alert_handler_sanity_vseq.sv [hunk 3]
```diff
@@ -81,88 +81,98 @@ class alert_handler_sanity_vseq extends alert_handler_base_vseq;
...
+
+      // if config is not locked, update max_intr_timeout and max_wait_phases cycles
+      if (!config_locked) begin
+        int max_intr_timeout_cyc;
+        foreach (intr_timeout_cyc[i]) begin
+          max_intr_timeout_cyc = (max_intr_timeout_cyc > intr_timeout_cyc[i]) ?
+                                  max_intr_timeout_cyc : intr_timeout_cyc[i];
+        end
+        max_wait_phases_cyc = (max_wait_phases_cyc > (max_phase_cyc * NUM_ESC_PHASES)) ?
+                               max_wait_phases_cyc : (max_phase_cyc * NUM_ESC_PHASES);
```
Comment (weicaiyang) [inline hw/ip/alert_handler/dv/env/seq_lib/alert_handler_sanity_vseq.sv:144]: nit: you can use `max2` to get the bigger value
./dv_utils/dv_utils_pkg.sv:  function automatic int max2(int a, int b);

File: hw/ip/alert_handler/dv/env/seq_lib/alert_handler_sanity_vseq.sv [hunk 4]
```diff
@@ -81,88 +81,98 @@ class alert_handler_sanity_vseq extends alert_handler_base_vseq;
...
+      // drive alert
+      drive_alert(alert_trigger, alert_int_err);
+
+      // if config is not locked, update max_intr_timeout and max_wait_phases cycles
+      if (!config_locked) begin
+        int max_intr_timeout_cyc;
+        foreach (intr_timeout_cyc[i]) begin
+          max_intr_timeout_cyc = (max_intr_timeout_cyc > intr_timeout_cyc[i]) ?
+                                  max_intr_timeout_cyc : intr_timeout_cyc[i];
+        end
```
Comment (weicaiyang) [inline hw/ip/alert_handler/dv/env/seq_lib/alert_handler_sanity_vseq.sv:142]: nit: you can max();
```
int max_q[$] = intr_timeout_cyc.max();
int max_intr_timeout_cyc = max_q[0];
```
FYI
https://stackoverflow.com/questions/47338777/why-does-system-verilog-max-and-min-functions-return-a-queue-and-not-a-singl

**Review by weicaiyang — APPROVED** [review]
LGTM with a few nits

File: hw/ip/alert_handler/dv/env/seq_lib/alert_handler_sanity_vseq.sv (hunk 4 above)
Comment (cindychip) [reply hw/ip/alert_handler/dv/env/seq_lib/alert_handler_sanity_vseq.sv]: thank you and thanks for the detailed reference, I will update the code to adopt this max function.

File: hw/ip/alert_handler/dv/env/seq_lib/alert_handler_sanity_vseq.sv (hunk 3 above)
Comment (cindychip) [reply hw/ip/alert_handler/dv/env/seq_lib/alert_handler_sanity_vseq.sv]: thanks Weicai for the pointer, updated.

File: hw/ip/alert_handler/dv/env/seq_lib/alert_handler_sanity_vseq.sv (hunk 2 above)
Comment (cindychip) [reply hw/ip/alert_handler/dv/env/seq_lib/alert_handler_sanity_vseq.sv]: thank you for the review. deleted the stall comments.

File: hw/ip/alert_handler/dv/tb/tb.sv (hunk 1 above)
Comment (cindychip) [reply hw/ip/alert_handler/dv/tb/tb.sv]: oh i forget to delete that. no it is not needed. Thanks
